events/messageReactionAdd.js

```diff
@@ -1,87 +1,71 @@
-          return true
-        }
-        var roles = config.Channelrole.some(role => {
-          if (message.guild.members.cache.find((member) => member.id === user.id).roles.cache.find(r => r.id === role)) {
-            return true
-          }
-        });
-        if (roles) {
-          return true
-        }
-        return false
+  if (user.bot) return; // Ignore bot's reactions
+
+  // Readfile collectors.json and save it in a variable
+  let collectors = JSON.parse(fs.readFileSync('./collectors.json', 'utf8'));
+
+  // Handle collectors
+  if (collectors.collector.filter((e) => e.id === messageReaction.message.id).length > 0) {
+    if (messageReaction.emoji.name === '📩') {
+      let channelname = `ticket-${user.username}`;
+      channelname = channelname.replace(/\s/g, '-').toLowerCase();
+      if (messageReaction.message.guild.channels.cache.find((channel) => channel.name === channelname) && config.one_app) {
+        user.send(`You already have an ongoing ticket.`).catch(console.error);
+        return messageReaction.users.remove(user.id);
       }
-    
-    function removeTicketfromCollectors(MessageID){
-        let collectors = JSON.parse(fs.readFileSync('./collectors.json', 'utf8'));
-        collectors.tickets = collectors.tickets.filter(e => e.id !== MessageID)
-        json = JSON.stringify(collectors);
-        fs.writeFile('./collectors.json', json, 'utf8', function(err) {
-            if (err) throw err;
-        });
+      contining(client, messageReaction.message, user);
+      messageReaction.users.remove(user.id);
     }
+  }
 
-    // Readfile collectors.json and save it in a variable
-    let collectors = JSON.parse(fs.readFileSync('./collectors.json', 'utf8'));
-    if (collectors.collector.filter(e => e.id === messageReaction.message.id).length > 0) {
-        if (user.bot) return;
-        switch (messageReaction.emoji.name) {
-            case "📩":{
-                let channelname = `ticket-${user.username}`
-                channelname = channelname.replace(/\s/g, '-').toLowerCase()
-                if (messageReaction.message.guild.channels.cache.find(channel => channel.name === channelname) && config.one_app) {
-                    user.send(`You already have an ongoing ticket.`).catch(console.error);
-                    messageReaction.users.remove(user.id)
-                    return;
-                } else {
-                    contining(client, messageReaction.message, user)
-                    messageReaction.users.remove(user.id)
-                }
-              }
-            }
+  // Handle tickets
+  if (collectors.tickets.filter((e) => e.id === messageReaction.message.id).length > 0) {
+    let channel = messageReaction.message.guild.channels.cache.find((channel) => channel.id === messageReaction.message.channel.id);
+    switch (messageReaction.emoji.name) {
+      case '🔒':
+        if (!checkUser(messageReaction.message, user) || !config.allow_user_lock) {
```

Comment:
Since this uses "!" you should use "&&" not "||"

Reply:
Oh, I see.

> (Allow users to lock their **own** tickets)